include/diracKernel/DiracSource.h

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// Moose Includes
+#include "DiracKernel.h"
+
+// Forward Declarations
+class DiracSource;
+
+template <>
+InputParameters validParams<DiracSource>();
+
+class DiracSource : public DiracKernel
+{
+public:
+  DiracSource(const InputParameters & parameters);
+
+  virtual void addPoints() override;
+  virtual Real computeQpResidual() override;
+
+protected:
+  Point _point;
+  int _dim;
```

Comment:
```suggestion
  /// Dimension of the mesh
  int _dim;
```
Also a one liner comment here (and for all the rest of the members).